compiler/src/main/scala/org/scalajs/core/compiler/PrepWasPublicBeforeTyper.scala

```diff
@@ -0,0 +1,72 @@
+/* Scala.js compiler
+ * Copyright 2013 LAMP/EPFL
+ * @author Nicolas Stucki
+ */
+
+package org.scalajs.core.compiler
+
+import scala.tools.nsc
+import nsc._
+
+/**
+ * @author Nicolas Stucki
+ */
+abstract class PrepWasPublicBeforeTyper extends plugins.PluginComponent with transform.Transform
+    with Compat210Component {
+
+  import global._
+
+  val phaseName = "prepwaspublicbeforetyper"
```

Comment:
How about: `jspretyper` (and the class then: `PreTyperComponent`). I think it is a good idea to keep the class name general. If we need more pre typer transforms, we won't add another phase, will we?